Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -7626,6 +7649,12 @@ void WebPageProxy::didSameDocumentNavigationForFrameViaJS(IPC::Connection& conne
     frame->didSameDocumentNavigation(WTFMove(url));
 
     protectedPageLoadState->commitChanges();
+#if ENABLE(WEBDRIVER_BIDI)
+    if (RefPtr automationSession = activeAutomationSession()) {
+        std::optional<WebCore::NavigationIdentifier> navID = navigation ? std::optional(navigation->navigationID()) : std::nullopt;
```

Comment:
This local variable seems unnecessary, since the type of the parameter _should_ be enough for the compiler to know to insert implicit constructors like std::optional(T). you should be able to inline the ternary expression, omit the std::optional constructor, use initializer braces, and/or use `auto` type to make it shorter.